### PR TITLE
added instructional resources submenu to student history row data

### PR DESCRIPTION
**Pull request by dsnyder-fairway**
screen shot 2017-12-13 at 2 28 39 pm

### Links*
screen shot 2017-12-13 at 2 28 45 pm

*note - 2 duplicate entries is likely the result of improper SQL update

### CODE REVIEW

File: webapp/src/main/webapp/src/app/assessments/popover/instructional-resource-popover.component.html [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<div *ngIf="resources && resources.length > 0; else noResources">
+<div *ngIf="resources && resources.length; else noResources">
```
Comment (dsnyder-fairway) [inline webapp/src/main/webapp/src/app/assessments/popover/instructional-resource-popover.component.html]: an improvement mentioned in #652

File: webapp/src/main/webapp/src/app/student/results/tables/student-history-iab-table.component.html [hunk 2]
```diff
@@ -50,7 +50,7 @@
         [popover]="buttonPopover"
         popoverTitle="{{ 'labels.groups.results.assessment.instruct-button' | translate:{ level:(('enum.iab-category.full.' + (exam.level ? exam.level.toString() : 'missing')) | translate) } }}"
         container="body"
-        (click)="loadInstructionalResources(rowData)"
+        (click)="loadInstructionalResources(data)"
```
Comment (dsnyder-fairway) [inline webapp/src/main/webapp/src/app/student/results/tables/student-history-iab-table.component.html]: while this was done in #652 I noticed I was having trouble with these links. This corrects it

File: webapp/src/main/webapp/src/app/student/results/tables/student-history-ica-summitive-table.component.html [hunk 3]
```diff
@@ -75,7 +75,7 @@
         [popover]="buttonPopover"
         popoverTitle="{{ 'labels.groups.results.assessment.instruct-button' | translate:{ level:(('enum.achievement-level.full.' + (exam.level ? exam.level.toString() : 'missing') ) | translate) } }}"
         container="body"
-        (click)="loadInstructionalResources(i)"
+        (click)="loadInstructionalResources(data)"
```
Comment (dsnyder-fairway) [inline webapp/src/main/webapp/src/app/student/results/tables/student-history-ica-summitive-table.component.html]: this change should have been done in #652

File: webapp/src/main/webapp/src/app/student/results/tables/student-history-iab-table.component.html [hunk 4]
```diff
@@ -6,7 +6,7 @@
   <!-- Menu column -->
   <p-column field="exam.date" header="{{'labels.student.results.cols.date' | translate}}" sortable="true">
     <ng-template let-data="rowData" let-exam="rowData.exam" pTemplate="body">
-      <popup-menu [item]="data" [actions]="actions" [text]="exam.date | date"></popup-menu>
+      <popup-menu [item]="data" [actions]="actions" [submenuItems]="loadOverallInstructionalResources(data)" [text]="exam.date | date"></popup-menu>
```
Comment (dsnyder-fairway) [inline webapp/src/main/webapp/src/app/student/results/tables/student-history-iab-table.component.html]: looking for a better word than `overall`. Suggestions?

File: webapp/src/main/webapp/src/styles.less [hunk 5]
```diff
@@ -387,4 +387,42 @@ body .select-menu-group {
   }
 }
 
-
+.dropdown-submenu {
```
Comment (dsnyder-fairway) [inline webapp/src/main/webapp/src/styles.less]: `dropdown-submenu` was a class in bootstrap v2. It was removed in bootstrap v3+

File: webapp/src/main/webapp/src/app/student/results/tables/student-history-iab-table.component.html (hunk 4 above)
Comment (wtritch) [reply webapp/src/main/webapp/src/app/student/results/tables/student-history-iab-table.component.html]: Does this front-load requesting resources?

Comment (wtritch) [reply webapp/src/main/webapp/src/app/student/results/tables/student-history-iab-table.component.html]: And to answer your question, maybe just `assessment` ?

Comment (dsnyder-fairway) [reply webapp/src/main/webapp/src/app/student/results/tables/student-history-iab-table.component.html]: yes - the `loadOverallInstructionalResources` is called when first going to a student test history report

Comment (dsnyder-fairway) [reply webapp/src/main/webapp/src/app/student/results/tables/student-history-iab-table.component.html]: in 00e1b53 we now load the resources on click

File: common-ngx/src/menu/popup-menu.component.ts [hunk 6]
```diff
@@ -24,18 +24,36 @@ import { Utils } from "../support/support";
...
+             triggers="{{ action.tooltip(item) == '' ? '': 'mouseenter:mouseleave'}}"
+             placement="right"
+             container="body">
+            {{action.displayName( item )}}
+          </a>
+          <ul class="dropdown-menu">
+            <li *ngFor="let submenu of submenuItems">
+              <a href="{{ submenu[0] }}" target="_blank">{{ submenu[ 1 ] }}</a>
+            </li>
+            <li *ngIf="submenuItems == undefined || submenuItems == null || submenuItems.length == 0">
```
Comment (dsnyder-fairway) [inline common-ngx/src/menu/popup-menu.component.ts]: this composite if-statement is required as I received an exception when checking `submenuItems.length == 0` and received no `'No resources found'` message when leaving this to `submenuItems == undefined || submenuItems == null`

**Comment by dsnyder-fairway** [comment]
image of "No resources found"
screen shot 2017-12-15 at 3 35 47 pm

**Pull request closed without merging**